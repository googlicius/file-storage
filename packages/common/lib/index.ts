export function common() {
  // TODO
  return 'this is common';
}
