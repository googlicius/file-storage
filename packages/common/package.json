{
  "name": "@file-storage/common",
  "version": "1.2.7",
  "description": "Common utilities, types, interfaces,... for `file-storage`.",
  "author": "Dang Nguyen <haidang009@gmail.com>",
  "homepage": "",
  "license": "MIT",
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "files": [
    "dist"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/googlicius/file-storage.git"
  },
  "publishConfig": {
    "registry": "https://registry.npmjs.org",
    "access": "public"
  },
  "scripts": {
    "build": "yarn clean && yarn compile",
    "clean": "rimraf -rf ./dist",
    "compile": "tsc -p tsconfig.build.json"
  },
  "dependencies": {
    "sharp": "^0.28.3"
  },
  "devDependencies": {
    "@types/sharp": "^0.28.5",
    "rimraf": "~3.0.2",
    "typescript": "~4.3.5"
  }
}
