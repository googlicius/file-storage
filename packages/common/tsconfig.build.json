{
  "extends": "../../tsconfig.build.json",
  "compilerOptions": {
    "outDir": "./dist"
  },
  "include": ["src/**/*.ts"],
  "exclude": ["**/*.spec.ts"]
}
