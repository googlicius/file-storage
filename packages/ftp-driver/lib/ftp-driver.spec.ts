import fs from 'fs';
import Storage from '@file-storage/core';
import { DriverName, getRootCwd } from '@file-storage/common';

describe('FTP Disk test', () => {
  beforeAll(() => {
    Storage.config({
      diskConfigs: [
        {
          driver: DriverName.FTP,
          name: 'sammy',
          root: '/upload',
          accessOptions: {
            host: '127.0.0.1',
            user: 'usertest',
            password: 'P@ssw0rd',
          },
        },
      ],
    });
  });

  test('Default disk is sammy', () => {
    expect(Storage.name).toEqual('sammy');
  });

  test('Upload image to ftp', async () => {
    const fileReadStream = fs.createReadStream(getRootCwd() + '/test/support/images/bird.jpeg');
    return expect(Storage.disk('sammy').put(fileReadStream, 'bird.jpeg')).resolves.toMatchObject({
      success: true,
      code: 226,
      message: '226 Transfer complete.',
    });
  });

  test('Upload ftp using Storage facade', () => {
    const fileReadStream = fs.createReadStream(getRootCwd() + '/test/support/images/bird.jpeg');
    return expect(Storage.put(fileReadStream, 'path/to/bird.jpeg')).resolves.toMatchObject({
      success: true,
      code: 226,
      message: '226 Transfer complete.',
      name: 'bird.jpeg',
      path: 'path/to/bird.jpeg',
    });
  });

  test('Upload ftp large image will uploaded to many formats', () => {
    const imageFileStream = fs.createReadStream(
      getRootCwd() + '/test/support/images/photo-1000x750.jpeg',
    );
    return expect(
      Storage.put(imageFileStream, 'my-photo/photo-1000x750.jpeg'),
    ).resolves.toMatchObject({
      success: true,
      code: 226,
      message: '226 Transfer complete.',
      name: 'photo-1000x750.jpeg',
      path: 'my-photo/photo-1000x750.jpeg',
      formats: {
        thumbnail: {
          name: 'thumbnail_photo-1000x750.jpeg',
          hash: null,
          ext: 'jpeg',
          mime: 'jpeg',
          width: 208,
          height: 156,
          size: 15.9,
          path: 'my-photo/thumbnail_photo-1000x750.jpeg',
        },
        large: {
          name: 'large_photo-1000x750.jpeg',
          hash: null,
          ext: 'jpeg',
          mime: 'jpeg',
          width: 1000,
          height: 750,
          size: 184.49,
          path: 'my-photo/large_photo-1000x750.jpeg',
        },
        medium: {
          name: 'medium_photo-1000x750.jpeg',
          hash: null,
          ext: 'jpeg',
          mime: 'jpeg',
          width: 750,
          height: 562,
          size: 105.12,
          path: 'my-photo/medium_photo-1000x750.jpeg',
        },
        small: {
          name: 'small_photo-1000x750.jpeg',
          hash: null,
          ext: 'jpeg',
          mime: 'jpeg',
          width: 500,
          height: 375,
          size: 52.33,
          path: 'my-photo/small_photo-1000x750.jpeg',
        },
      },
    });
  });

  test('Download image from ftp', async () => {
    const fileReadStream = fs.createReadStream(getRootCwd() + '/test/support/images/bird.jpeg');
    await Storage.disk('sammy').put(fileReadStream, 'test_upload/bird.jpeg');

    return expect(Storage.disk('sammy').get('test_upload/bird.jpeg')).resolves.toBeTruthy();
  });

  test('Delete image from ftp', async () => {
    const fileReadStream = fs.createReadStream(getRootCwd() + '/test/support/images/bird.jpeg');
    await Storage.disk('sammy').put(fileReadStream, 'test_upload/bird.jpeg');

    return expect(Storage.disk('sammy').delete('test_upload/bird.jpeg')).resolves.toMatchObject({
      code: 250,
      message: '250 Delete operation successful.',
    });
  });

  test('File is exists', async () => {
    const fileReadStream = fs.createReadStream(getRootCwd() + '/test/support/images/bird.jpeg');
    await Storage.disk('sammy').put(fileReadStream, 'test_upload/bird.jpeg');

    return expect(Storage.exists('test_upload/bird.jpeg')).resolves.toEqual(true);
  });

  test('File is not exists', async () => {
    return expect(Storage.disk('sammy').exists('not-exists.jpeg')).resolves.toEqual(false);
  });

  test('Get file size', async () => {
    const fileReadStream2 = fs.createReadStream(getRootCwd() + '/test/support/images/bird.jpeg');
    await Storage.disk('sammy').put(fileReadStream2, 'bird-images/bird.jpeg');

    return expect(Storage.size('bird-images/bird.jpeg')).resolves.toEqual(56199);
  });

  test('Last modified', async () => {
    const fileReadStream = fs.createReadStream(getRootCwd() + '/test/support/images/bird.jpeg');
    await Storage.disk('sammy').put(fileReadStream, 'bird-images/bird2.jpeg');
    const lastMod = await Storage.lastModified('bird-images/bird2.jpeg');
    expect(typeof lastMod).toBe('number');
  });
});
