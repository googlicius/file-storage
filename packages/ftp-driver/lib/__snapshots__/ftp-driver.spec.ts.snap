// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FTP Disk test Upload ftp large image will uploaded to many formats 1`] = `
{
  "code": 226,
  "formats": {
    "large": {
      "ext": "jpeg",
      "hash": null,
      "height": 750,
      "mime": "jpeg",
      "name": "large_photo-1000x750.jpeg",
      "path": "my-photo/large_photo-1000x750.jpeg",
      "size": 184.49,
      "width": 1000,
    },
    "medium": {
      "ext": "jpeg",
      "hash": null,
      "height": 562,
      "mime": "jpeg",
      "name": "medium_photo-1000x750.jpeg",
      "path": "my-photo/medium_photo-1000x750.jpeg",
      "size": 107.23,
      "width": 750,
    },
    "small": {
      "ext": "jpeg",
      "hash": null,
      "height": 375,
      "mime": "jpeg",
      "name": "small_photo-1000x750.jpeg",
      "path": "my-photo/small_photo-1000x750.jpeg",
      "size": 52.4,
      "width": 500,
    },
    "thumbnail": {
      "ext": "jpeg",
      "hash": null,
      "height": 156,
      "mime": "jpeg",
      "name": "thumbnail_photo-1000x750.jpeg",
      "path": "my-photo/thumbnail_photo-1000x750.jpeg",
      "size": 16.14,
      "width": 208,
    },
  },
  "message": "226 Transfer complete.",
  "name": "photo-1000x750.jpeg",
  "path": "my-photo/photo-1000x750.jpeg",
  "success": true,
}
`;

exports[`FTP Disk test append should append a text to a file 1`] = `
"First line
Appended line"
`;
